akka-docs-dev/rst/scala/code/docs/stream/cookbook/RecipeReduceByKey.scala

```diff
@@ -54,15 +47,17 @@ class RecipeReduceByKey extends RecipeSpec {
         groupKey: (In) => K,
         foldZero: (K) => Out)(fold: (Out, In) => Out): Flow[In, (K, Out), Unit] = {
 
-        val groupStreams = Flow[In].groupBy(groupKey)
-        val reducedValues = groupStreams.map {
-          case (key, groupStream) =>
-            groupStream.runFold((key, foldZero(key))) {
-              case ((key, aggregated), elem) => (key, fold(aggregated, elem))
-            }
-        }
-
-        reducedValues.buffer(maximumGroupSize, OverflowStrategy.fail).mapAsync(4)(identity)
+        Flow[In]
+          .groupBy(maximumGroupSize, groupKey)
+          .fold(Option.empty[(K, Out)]) {
+            case (None, elem) =>
+              val key = groupKey(elem)
+              Some((key, fold(foldZero(key), elem)))
+            case (Some((key, out)), elem) =>
+              Some((key, fold(out, elem)))
+          }
+          .map(_.get)
```

Comment:
Yes, if you re-read what I wrote, you'll notice that I have nothing against throwing exceptions but what I was against was A) the use of Option.get and B) the generated exception from using Option.get

Option.get should not be encouraged, and in this case the error message ought to be different than: java.util.NoSuchElementException: None.get